tedge/src/system_services/factory.rs

```diff
@@ -0,0 +1,31 @@
+use crate::system_services::*;
+use crate::utils::users::UserManager;
+
+pub trait SystemServiceManagerFactory {
+    fn create(&self) -> Box<dyn SystemServiceManager>;
+}
+
+pub struct DefaultSystemServiceManagerFactory {
+    user_manager: UserManager,
+}
+
+impl DefaultSystemServiceManagerFactory {
+    pub fn new(user_manager: UserManager) -> Self {
+        Self { user_manager }
+    }
+}
+
+#[cfg(all(feature = "systemd", feature = "openrc"))]
+compile_error!("Both features \"systemd\" and \"openrc\" cannot be enabled at the same time.");
```

Comment:
You anticipated a question I got reading the feature list!

Yes, it makes sense to have `tedge` compiled for a single init system. But in that case why a factory? Why not __a__ Service set at compile time?